test/simple_malloc.cc

```diff
@@ -126,9 +128,19 @@ extern "C" DDPROF_NOINLINE void do_lot_of_allocations(const Options &options,
     // NOLINTNEXTLINE(clang-analyzer-unix.Malloc)
     DoNotOptimize(p2);
 
-    if (skip_free++ >= options.skip_free) {
-      free(p2);
-      skip_free = 0;
+    if (options.keep_live_allocations > 0) {
+      if (p2 != nullptr) {
+        live_allocations.push_back(p2);
+        while (live_allocations.size() > options.keep_live_allocations) {
+          free(live_allocations.front());
+          live_allocations.pop_front();
+        }
+      }
+    } else {
+      if (skip_free++ >= options.skip_free) {
```

Comment:
maybe I remove the skip free ? I don't like skip free it ends up killing my laptop :smile:

Reply:
skip free is useful to check that we are able to detect leaks